src/Foundatio.Parsers.LuceneQueries/Visitors/AssignOperationTypeVisitor.cs

```diff
@@ -9,11 +9,19 @@ public class AssignOperationTypeVisitor : ChainableQueryVisitor
 {
     public override Task VisitAsync(GroupNode node, IQueryVisitorContext context)
     {
+        if (node.HasOperationType())
+            return Task.CompletedTask;
+
         if (String.IsNullOrEmpty(node.Field))
             return base.VisitAsync(node, context);
 
         if (node.Left is not TermNode leftTerm)
         {
+            // For sub aggregations we need to see if there is a parent with parens
+            var closestParentWithParens = node.GetGroupNode();
```

Comment:
This could be a root node

Reply:
You should add an overload to GetGroupNode to walk up the tree until it finds a groupnode with parens.

Reply:
https://github.com/FoundatioFx/Foundatio.Parsers/blob/main/src/Foundatio.Parsers.LuceneQueries/Extensions/QueryNodeExtensions.cs#L146-L161